src/vision/shiny_future/alan_switches_runtimes.md

```diff
@@ -0,0 +1,148 @@
+
+Alan says, "And so, if I want to get things running on `humboldt` again, I spawn a task back on `humboldt`?"
+
+"Exactly," says Barbara. "Humboldt has a global event loop, so you can do that by just doing `humboldt::spawn`. You can also just use the `humboldt::io` APIs directly and so forth if you want to do I/O with humboldt."
+
+Alan winds up with some code that looks like this:
+
+```rust
+async fn do_something_on_humboldt() {
+    gcd::spawn(async move {
+        let foo = do_something_on_gcd();
+
+        let bar = humboldt::spawn(async move {
+            do_a_little_bit_of_stuff_on_humboldt();
+        });
+
+        combine(foo.await, bar.await);
+    });
+}
+```
+
+## 🤔 Frequently Asked Questions
+
+### What status quo story or stories are you retelling?
+
+Good question! I'm not entirely sure! I have to go looking and think about it. Maybe we'll have to write some more.
+
+### What are the key points you were trying to convey with this status quo story?
+
+* There is some way to seamlessly change to a different default runtime to use for `async main`.
+* There is no global runtime, just the current runtime.
+* When you are using this different runtime, you can write code that is hard-coded to it and which exposes additional capabilities.
+* You can integrate multiple runtimes relatively easily, and the std APIs work with whechever is the current runtime.
+
+### How do you imagine the std APIs and so forth know the current runtime?
+
+I was imagining that we would add fields to the `Context<'_>` struct that is supplied to each `async fn` when it runs. Users don't have direct access to this struct, but the compiler does. If the std APIs return futures, they would gain access to it that way as well. If not, we'd have to create some other mechanism.
+
+### What happens for runtimes that don't support all the features that std supports?
+
+That feels like a portability question. See the (yet to be written) sequel story, "Alan runs some things on WebAssembly". =)
+
+### **What is [Alan] most excited about in this future? Is he disappointed by anything?**
+
+Alan is excited about how easy it is to get async programs up and running, and he finds that they perform pretty well once he does so, so he's happy.
+
+### **What is [Grace] most excited about in this future? Is she disappointed by anything?**
+
+Grace is concerned with memory safety and being able to deploy her tricks she knows from other languages. Memory safety works fine here. In terms of tricks she knows and loves, she's happy that she can easily switch to another runtime. The default runtime is good and works well for most things, but for the [`DistriData`] project, they really need something tailored just for them. She is also happy she can use the extended APIs offered by `humboldt`.
```

Comment:
Grace would probably be concerned the overhead that is produced by all the traits, and stuff which might be important n certain embedded environments. Similarly to Barbara but specific to embedded environments.

Reply:
In what way do traits introduce overhead? Do you mean because of dynamic dispatch, or do you mean because having to "fit oneself" to a trait may induce overhead?

Reply:
I assume that going through trait objects and probably even some generics, will generate much more code which could lead to exceeding some size limit on embedded devices. It may not even be true in reality but I thought it could be.

Reply:
I don't know that it's true, but it is plausible. I did add some notes about embedded devices in the FAQ -- in terms of what sorts of projects would be affected -- I can add some comments to this effect as well.

Reply:
Done.